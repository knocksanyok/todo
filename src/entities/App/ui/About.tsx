import { useNavigate } from 'react-router'
import Button from '@mui/material/Button'

const About = ({ title, version, onClick }: { title: string; version: string; onClick: () => void }) => {
	// const navigate = useNavigate()
	return (
		<div style={{ backgroundColor: 'violet' }}>
			<h1>{title}</h1>
			<span>{version}</span>
			<Button
				onClick={() => {
					onClick()
				}}
			>
				Жмякай
			</Button>
			{/*<Button variant={'contained'} onClick={() => navigate(-1)}>*/}
			{/*	Go Back*/}
			{/*</Button>*/}
		</div>
	)
}

export default About
