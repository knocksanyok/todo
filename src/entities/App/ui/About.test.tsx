import { cleanup, fireEvent, render, screen } from '@testing-library/react'
import About from './About.tsx'
import { describe, test, afterEach, expect } from 'vitest'

describe('About component', () => {
	afterEach(cleanup)

	test('renders correctly', () => {
		const rendered = render(<About />)
		expect(rendered.baseElement).toMatchSnapshot()
		screen.debug()
		const heading = screen.getByTestId('version-container')
		expect(heading).toBeDefined()
	})
	test('increment counter', () => {
		render(<About />)
		const count = screen.getByTestId('counter')
		const countContent = count.textContent
		expect(countContent).toBe('0')
		expect(countContent).not.toBeNull()

		const button = screen.getByTestId('increment-button')
		fireEvent.click(button)

		const newCount = screen.getByTestId('counter')
		const newCountContent = newCount.textContent
		expect(newCountContent).toBe('1')
		expect(newCountContent).not.toBeNull()
		expect(newCountContent).not.toBe(countContent)
	})
})
