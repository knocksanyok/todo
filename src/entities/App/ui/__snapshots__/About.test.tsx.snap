// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About component > renders correctly 1`] = `
<body>
  <div>
    <div>
      <h1>
        About Us
      </h1>
      <span
        data-testid="version-container"
      >
        Version 1.0.0
      </span>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
        data-testid="increment-button"
        tabindex="0"
        type="button"
      >
        Click
      </button>
      <span
        data-testid="counter"
      >
        0
      </span>
    </div>
  </div>
</body>
`;
